src/activities/assignments/AssignmentActivityEntity.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+import { Entity } from '../../es6/Entity.js';
+
+/**
+ * AssignmentActivityEntity class representation of a d2l AssignmentActivity.
+ */
+export class AssignmentActivityEntity extends Entity {

```

Comment:
I'm probably being too literal/strict here, but wouldn't this be `AssignmentEntity`, since we're outside of the `activities` HM domain at this point? (Although I recognize that all of this is kind of within the activities domain, in a way... 🤔)

Reply:
Yeh - I ummed and ahhed over the naming... i don't feel strongly either way... an assignment is an activity but would we also have a representation of an AssignmentActivity that is not an AssignmentActivityUsage but is also not the core Assignment. Maybe. Eric's original model had ActivityTemplate, ActivityInstance (which seems to have become ActivityUsage?), and AssignedActivity. So maybe if we had an entity that was not a usage but was also just the activity common stuff that would be an AssignmentActivityEntity.

I think I'll rename it to AssignmentEntity just because it's shorter!

Reply:
ActivityUsage I believe is the old terminology, whereas ActivityTemplate/Instance/Assigned are the new ones, so old -> new mapping becomes

Activity -> ActivityTemplate
ActivityUsage -> ActivityInstance
UserActivityUsage -> AssignedActivity

I do like these names better tbh, sadly the codebase is full of the old ones :/